File: TheaterSuggestions/CSharp/SeatsSuggestions/Model/SeatWithDistance.cs
using System.Collections.Generic;
using Value;

namespace SeatsSuggestions.Model
{
    public class SeatWithDistance : ValueType<Seat>
    {
        public SeatWithDistance(Seat seat, int distanceFromTheMiddle)
        {
            Seat = seat;
            DistanceFromTheMiddle = distanceFromTheMiddle;
        }

        public Seat Seat { get; }
        public int DistanceFromTheMiddle { get; }

        public override string ToString()
        {
            return $"{Seat.RowName}{Seat.Number}";
        }

        protected override IEnumerable<object> GetAllAttributesToBeUsedForEquality()
        {
            return new object[] { Seat, DistanceFromTheMiddle };
        }
    }
}

File: TheaterSuggestions/Java/SeatsSuggestions/src/test/java/com/baasie/SeatsSuggestions/SeatWithDistance.java
package com.baasie.SeatsSuggestions;
import lombok.EqualsAndHashCode;
import com.baasie.SeatsSuggestions.Seat;

@EqualsAndHashCode
public class SeatWithDistance {
    private final Seat seat;
    private final int distanceFromTheMiddleOfTheRow;

    public SeatWithDistance(Seat seat, int distanceFromTheMiddleOfTheRow) {
        this.seat = seat;
        this.distanceFromTheMiddleOfTheRow = distanceFromTheMiddleOfTheRow;
    }

    public Seat seat() {
        return this.seat;
    }

    public int distanceFromTheMiddleOfTheRow() {
        return distanceFromTheMiddleOfTheRow;
    }
}
